pkg/util/appgroup.go

```diff
@@ -0,0 +1,316 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/scheduler-plugins/apis/scheduling/v1alpha1"
+	"sort"
+	"strings"
+)
+
+// Sort AppGroupTopologyList by Workload.Selector
+type ByWorkloadSelector v1alpha1.AppGroupTopologyList
+
+func (s ByWorkloadSelector) Len() int {
+	return len(s)
+}
+
+func (s ByWorkloadSelector) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (s ByWorkloadSelector) Less(i, j int) bool {
+	return s[i].Workload.Selector < s[j].Workload.Selector
+}
+
+// Sort AppGroupWorkloadList by Workload.Selector
+type BySelector v1alpha1.AppGroupWorkloadList
+
+func (s BySelector) Len() int {
+	return len(s)
+}
+
+func (s BySelector) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func (s BySelector) Less(i, j int) bool {
+	return s[i].Workload.Selector < s[j].Workload.Selector
+}
+
+// FindWorkloadByName : return the workload's Info based on its name
+func FindWorkloadByName(workloadList v1alpha1.AppGroupWorkloadList, name string) v1alpha1.AppGroupWorkloadInfo {
+	low := 0
+	high := len(workloadList) - 1
+
+	for low <= high {
+		mid := (low + high) / 2
+		if workloadList[mid].Workload.Name == name {
+			return workloadList[mid].Workload // Return the WorkloadInfo
+		} else if workloadList[mid].Workload.Name < name {
+			low = mid + 1
+		} else if workloadList[mid].Workload.Name > name {
+			high = mid - 1
+		}
+	}
+	// Workload Info was not found
+	return v1alpha1.AppGroupWorkloadInfo{}
+}
+
+// GetPodAppGroupLabel : get AppGroup from pod annotations
+func GetPodAppGroupLabel(pod *v1.Pod) string {
+	return pod.Labels[v1alpha1.AppGroupLabel]
+}
+
+// GetPodAppGroupSelector : get Workload Selector from pod annotations
+func GetPodAppGroupSelector(pod *v1.Pod) string {
```

Comment:
This function is currently unused

Reply:
This function is used by the plugins to get the AppGroup label from pods. Indeed, it is unused since the plugins are not included in this PR...